### ISSUE TITLE
Fedora packaging

### ISSUE THREAD

**Issue by Sinthoras7** [sezanzeb/input-remapper#537]
There are only instructions on how to install it on Debian and Arch based systems, not Fedora.
 Can it be installed on Fedora too? If so, how?

**Comment by sezanzeb** [comment]
Try the manual installation as shown in the readme. I don't use fedora, so I can't offer support.

**Comment by sezanzeb** [comment]
Ideally someone who knows how this works gets input-remapper into fedoras official repo https://docs.fedoraproject.org/en-US/package-maintainers/New_Package_Process_for_New_Contributors/

Or contributes a script that creates a .rpm package file

**Comment by Tesla-Tank** [comment]
> There are only instructions on how to install it on Debian and Arch based systems, not Fedora. Can it be installed on Fedora too? If so, how?

Try using this copr, I've used it for a while now and had no issues installing or using it. 
https://copr.fedorainfracloud.org/coprs/sunwire/input-remapper/

**Comment by Zeit08** [comment]
Thanks, I also was looking for input-remapper for Fedora, and I also suggest it to be in the main repo as it may be a fundamental program for some of us. Unfortunately I have no skill in compiling and maintaining it in the repo...

**Comment by sezanzeb** [comment]
> First, the .py files that do not end up in /usr/bin but reside in the system's site packages directory (e.g. /usr/lib/python3.11/site-packages/) should not have a shebang line (#!/usr/bin/python3)

Thanks. I Just removed them from both main and beta

> input-remapper

that would be the intended name

> inputremapper

Because python packages can't have dashes in their name. `import input-remapper` is not valid python syntax unfortunately

> key-mapper

The old name

> input_remapper

Only seen in the code to comply with the snake_case Python convention for variable names

**Comment by alxpl** [comment]
Thanks, it's been so long since I last touched python code that I had completely pushed out of my memory its naming conventions.
I will get in touch with Paweł who has set up the COPR and see if he's willing to (co-)maintain the package in Fedora.
When do you think the 1.6 release will be ready?
I am a bit swamped with work at the moment, I haven't even had the time to deal with #542. This is going to be a weekend project, but once the package is up for review, things will move along reasonably fast, there's quite a few people who "owe" me package reviews.

**Comment by sezanzeb** [comment]
It will be a 2.0 release, and probably in february

**Comment by alxpl** [comment]
Hello,

I've started working on this again, pretty much everything is in place, all that's left is an AppStream metadata file. Do you intend to provide one? I can whip up one for you, but it would be better if at least the screenshots were hosted someplace you control and you may want to specify things like the id yourself.

https://www.freedesktop.org/software/appstream/docs/chap-Quickstart.html
In order to capture the entire application window at a 16:9 ratio, there is a handy GNOME extension:
https://extensions.gnome.org/extension/881/screenshot-window-sizer/

**Comment by sezanzeb** [comment]
Are the screenshots good like that? I'm running kde, so I just tried to resize it to be close to 16:9

**Comment by alxpl** [comment]
> Are the screenshots good like that? I'm running kde, so I just tried to resize it to be close to 16:9

I ran appstream-util with 'validate-strict' and it did not complain about them, so they're good.


> 2.0.0-rc is on main. I'll make the proper 2.0.0 release in two weeks.

I will try to find some time to work on an updated package and I'll get back to you if I have any issues.

**Comment by alxpl** [comment]
I ended up with several duplicate files and I am trying to figure out if it's something in setup.py or in my spec file. Specifically, rpmlint warns about these files:

```
python3-input-remapper.noarch: W: files-duplicate /usr/share/input-remapper/input-remapper-gtk.desktop /usr/share/applications/input-remapper-gtk.desktop
python3-input-remapper.noarch: W: files-duplicate /usr/share/polkit-1/actions/input-remapper.policy /usr/share/input-remapper/input-remapper.policy
python3-input-remapper.noarch: W: files-duplicate /usr/share/metainfo/io.github.sezanzeb.input_remapper.metainfo.xml /usr/share/input-remapper/io.github.sezanzeb.input_remapper.metainfo.xml
python3-input-remapper.noarch: W: files-duplicate /usr/share/input-remapper/lang/fr_FR/LC_MESSAGES/input-remapper.mo /usr/share/input-remapper/lang/fr/LC_MESSAGES/input-remapper.mo
python3-input-remapper.noarch: W: files-duplicate /usr/share/input-remapper/lang/it_IT/LC_MESSAGES/input-remapper.mo /usr/share/input-remapper/lang/it/LC_MESSAGES/input-remapper.mo
python3-input-remapper.noarch: W: files-duplicate /usr/share/input-remapper/lang/ru_RU/LC_MESSAGES/input-remapper.mo /usr/share/input-remapper/lang/ru/LC_MESSAGES/input-remapper.mo
python3-input-remapper.noarch: W: files-duplicate /usr/share/input-remapper/lang/sk_SK/LC_MESSAGES/input-remapper.mo /usr/share/input-remapper/lang/sk/LC_MESSAGES/input-remapper.mo
python3-input-remapper.noarch: W: files-duplicate /usr/share/input-remapper/lang/uk_UA/LC_MESSAGES/input-remapper.mo /usr/share/input-remapper/lang/uk/LC_MESSAGES/input-remapper.mo
python3-input-remapper.noarch: W: files-duplicate /usr/share/input-remapper/lang/zh_CN/LC_MESSAGES/input-remapper.mo /usr/share/input-remapper/lang/zh/LC_MESSAGES/input-remapper.mo

```

Is there any chance this is the result of globbing in the data dir in line 116 of setup.py?

**Comment by sezanzeb** [comment]
> Is there any chance this is the result of globbing in the data dir in line 116 of setup.py?

I think so too

**Comment by alxpl** [comment]
I have submitted the package for review, I will let you know if anything else is required before it is accepted.

**Comment by baughmann** [comment]
I just wanted to say thank you to you all! Saving me a lot of time and headache. LogiOps decided to break on me.

**Comment by alxpl** [comment]
Hello everyone,

Apologies for the delay, input-remapper is now officially in Fedora. Within a few hours it should be available for rawhide users and in the updates-testing repositories for F38 and F37.

It was suggested during the review that I add a small notice explaining to users what they need to do to get the package working, which I've added [here](https://src.fedoraproject.org/rpms/input-remapper/raw/rawhide/f/README.Fedora).

There are some minor issues with duplicate files and file locations, but I've dealt with them at the package level, we can go over them at a later time.

**Comment by bayazidbh** [comment]
So we can now remove sunwire/input-remapper copr and can just specify `input-remapper` with dnf?

**Comment by alxpl** [comment]
> So we can now remove sunwire/input-remapper copr and can just specify `input-remapper` with dnf?

Yes, but since the two packages don't have the same name (python-input-remapper vs input-remapper), you will have to remove the one from COPR before installing the package from the Fedora repo. The former was never in Fedora, so I couldn't add an "Osoletes" directive to provide an upgrade path.

**Comment by bayazidbh** [comment]
How does the autoload works in the fedora repo version? `python-input-remapper` from the copr was able to autoload my mouse just fine, once I toggled it, but the official `input-remapper` version don't.

Might be an issue with immutable distro (I use ublue-os, installed input-remapper to my image) as I have had issues with samba (installed on image directly on github) and teamviewer (installed through rpm-install) not having their systemd services installed in the right place.

**Comment by alxpl** [comment]
On Wed, Jun 21, 2023 at 3:32 PM Bayazid B. Halim ***@***.***>
wrote:

> How does the autoload works in the fedora repo version?
> python-input-remapper from the copr was able to autoload my mouse just
> fine, once I toggled it, but the official input-remapper version don't.
>

According to the guidelines, if a service requires some sort of
configuration to function properly, it cannot be enabled by default:
https://docs.fedoraproject.org/en-US/packaging-guidelines/DefaultServices/#_must_not_require_manual_configuration_to_function

Have you enabled the input-remapper service as documented here?
https://github.com/sezanzeb/input-remapper#fedora

**Comment by bayazidbh** [comment]
Okay, that works, thanks @alxpl
